go/extractor/cli/go-autobuilder/go-autobuilder.go

```diff
@@ -719,82 +719,150 @@ func installDependenciesAndBuild() {
 const minGoVersion = "1.11"
 const maxGoVersion = "1.20"
 
+// Check if `version` is lower than `minGoVersion`. Note that for this comparison we ignore the
+// patch part of the version, so 1.20.1 and 1.20 are considered equal.
+func belowSupportedRange(version string) bool {
+	return semver.Compare(semver.MajorMinor("v"+version), "v"+minGoVersion) < 0
+}
```

Comment:
I keep thinking that I really wish Go had something like Haskell's `newtype` which let you create a new nominal type to distinguish e.g. between version strings that include the `"v"` and ones that don't.

Reply:
I don't know Haskell, but perhaps you can create a new type with the same underlying type with `type semverVersion string`.

I admit that the treatment of version strings in this file is a mess. I've stuck with no prefix in the code I've added, but I haven't changed it elsewhere, which just stores up the problem for later. I wanted to avoid accidentally changing current behaviour.